frontend/src/state/GraphState.jsx

```diff
@@ -427,79 +428,86 @@ export class GraphContainer {
     let displayOrder = 0;
     let rows = {
       displayedRows: {
-        name: "Visible rows", displayOrder: displayOrder++,
+        name: "Visible rows",
+        displayOrder: displayOrder++,
         value: displayedConcepts.length,
       },
       concepts: {
-        name: "Concepts", displayOrder: displayOrder++,
-        value: this.gd.concept_ids.length,
-        hiddenConceptCnt: setOp('difference', this.gd.concept_ids, displayedCids).length,
-        displayedConceptCnt: setOp('intersection', this.gd.concept_ids, displayedCids).length,
+        name: "Concepts",
+        displayOrder: displayOrder++,
+        value: this.gd.concept_ids?.length ?? 0,
+        hiddenConceptCnt: setOp('difference', this.gd.concept_ids ?? [], displayedCids).length,
+        displayedConceptCnt: setOp('intersection', this.gd.concept_ids ?? [], displayedCids).length,
         specialTreatmentRule: 'expand all',
         specialTreatmentDefault: false,
       },
       definitionConcepts: {
-        name: "Definition concepts", displayOrder: displayOrder++,
-        value: this.gd.specialConcepts.definitionConcepts.length,
-        hiddenConceptCnt: setOp('difference', this.gd.specialConcepts.definitionConcepts, displayedCids).length,
+        name: "Definition concepts",
+        displayOrder: displayOrder++,
+        value: this.gd.specialConcepts.definitionConcepts?.length ?? 0,
+        hiddenConceptCnt: setOp('difference', this.gd.specialConcepts.definitionConcepts ?? [], displayedCids).length,
```

Comment:
## Bug fix Potentially undefined special concepts types
I [asked ChatGPT](https://chatgpt.com/share/64697079-8e8f-460d-8006-ee4040341645) to handle cases of potentially undefined `PROPERTY` in `this.gd.specialConcepts.PROPERTY.length?` patterns.

@Sigfried I think that these are undefined because the code in `develop` is unfinished. I ran the test on the `add-concepts` branch, and though it did error out in some cases, it got farther than on `develop`, indicating to me that you have gotten further in your development of this part of handling special concepts.

@Sigfried I leave it up to you if you want to keep this change or revert. I think that when you've finished with the special concepts, this will no longer be necessary. It makes the code harder to understand, but it is also potentially useful edge case handling, though you can say much better than I how useful it is. I just know for now that without these changes, the Jest tests will fail.

Reply:
Siggie showed me that `lodash` has a much easier to read solution, e.g.:
`get(this, "gd.concept_ids") || []`